sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -68,7 +68,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] {
    * generating case class `equals` methods
    */
   def fastEquals(other: TreeNode[_]): Boolean = {
-    this.eq(other) || this == other
+    this.eq(other)// || this == other
```

Comment:
I'm a little nervous about this change. Before it, we can create a same tree node(not return the origin one) during tree node transformation which will be regarded as no-change and finally make the batch reach fixed point. However, we can't do this now and may make a batch exceed max iterations which will slow down batch execution dramatically.
cc @marmbrus